frontend/src/components/FundsPullPreApproval/FundsPullPreApprovalsList.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState } from "react";
+import { Approval } from "../../interfaces/approval";
+import ApproveModal from "./ApproveModal";
+import RejectModal from "./RejectModal";
+import FundsPullPreApproval from "./FundsPullPreApproval";
+import RevokeModal from "./RevokeModal";
+
+interface ApprovalsListProps {
+  approvals: Approval[];
+  displayApproveRejectButtons: boolean;
+  displayRevokeButton: boolean;
+}
+
+function FundsPullPreApprovalsList({
+  approvals,
+  displayApproveRejectButtons,
+  displayRevokeButton,
+}: ApprovalsListProps) {
+  const [approveModalOpen, setApproveModalOpen] = useState<boolean>(false);
+  const [rejectModalOpen, setRejectModalOpen] = useState<boolean>(false);
+  const [revokeModalOpen, setRevokeModalOpen] = useState<boolean>(false);
+  const [approvalInModal, setApprovalInModal] = useState<Approval>();
+
+  return (
+    <>
+      {approvals.map((approval) => {
+        return (
+          <FundsPullPreApproval
+            key={approval.funds_pull_pre_approval_id}
+            approval={approval}
+            onApproveClick={() => setApproveModalOpen(true)}
+            onRejectClick={() => setRejectModalOpen(true)}
+            onRevokeClick={() => setRevokeModalOpen(true)}
+            onAnyClickSetApproval={() => setApprovalInModal(approval)}
```

Comment:
I would merge this method into each of the action methods above (approve, reject, revoke). It's a bit confusing when to use this method and not a common practice.